node.cpp

```diff
@@ -0,0 +1,192 @@
+#include <mutex>
+
+#include "node.h"
+#include "roles.h"
+
+std::once_flag init_once, cleanup_once;
+
+void Node::initialize()
+{
+  std::call_once(init_once, []() { xmlInitParser(); });
+}
+
+void Node::cleanup()
+{
+  std::call_once(cleanup_once, []() { xmlCleanupParser(); });
+}
+
+Node::Node(std::string internal_type) : internal_type(internal_type), parent(NULL), xpathCtx(NULL), doc(NULL) {}
+
+Node::~Node()
+{
+  printf("DEALLOCATED NODE\n");
```

Comment:
sure! but i think it is good for the early moments of the library. if it crash in random ways, having this print here always enabled can help a lot to fix it.